### PR TITLE
fix: retry

### PR DESCRIPTION
**Pull request by CristhianF7**
## Description

- Handles secret creation error properly `secrets "container-registry-auth" already exists`
- Added execution control for post-detokenize process

### CODE REVIEW

File: cmd/k3d/create.go [hunk 1]
```diff
@@ -1187,34 +1187,39 @@ func runK3d(cmd *cobra.Command, _ []string) error {
...
-			return fmt.Errorf("failed to rename remote-backend.md to remote-backend.tf: %w", err)
+			return fmt.Errorf("error opening repo at %q: %w", config.GitopsDir, err)
+		}
+		_, err = os.Stat(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider))
+		if err == nil {
+			err = os.Rename(fmt.Sprintf("%s/terraform/%s/remote-backend.md", config.GitopsDir, config.GitProvider), fmt.Sprintf("%s/terraform/%s/remote-backend.tf", config.GitopsDir, config.GitProvider))
+			if err != nil {
+				return fmt.Errorf("failed to rename remote-backend.md to remote-backend.tf: %w", err)
+			}
 		}
```
Comment (patrickdappollonio) [inline cmd/k3d/create.go:1206]: This seems like the opposite of what you want to do and it's considered an anti-pattern in Go: not only it eats the error, it also handles the non-error like an error.

If you can't stat the file, do you still continue?

File: internal/gitShim/containerRegistryAuth.go [hunk 2]
```diff
@@ -48,12 +49,20 @@ func CreateContainerRegistrySecret(obj *ContainerRegistryAuth) (string, error) {
...
 		}
-		if err := k8s.CreateSecretV2(obj.Clientset, argoDeployTokenSecret); err != nil {
+		err := k8s.CreateSecretV2(obj.Clientset, argoDeployTokenSecret)
+		if errors.IsAlreadyExists(err) {
+			if err := k8s.UpdateSecretV2(obj.Clientset, "argo", secretName, data); err != nil {
+				return "", fmt.Errorf("error while updating secret for GitHub container registry auth: %w", err)
+			}
+		}
+
+		if err != nil {
```
Comment (patrickdappollonio) [inline internal/gitShim/containerRegistryAuth.go:65]: You might wanna check too for...

```suggestion
		if err != nil && !errors.IsAlreadyExists(err) {
```

The `errors.IsAlreadyExists()` has a tree: if it already exists and you updated the flow successfully, `err` will still be true to `errors.IsAlreadyExists()` when you hit this line. The `err` on `UpdateSecretV2` is scoped to the `if`.

File: cmd/k3d/create.go (hunk 1 above)
Comment (CristhianF7) [reply cmd/k3d/create.go:1206]: I didn't write this code, I only wrapped it in a condition so it doesn't run again for retries. Other than that I Agree

File: internal/gitShim/containerRegistryAuth.go (hunk 2 above)
Comment (CristhianF7) [reply internal/gitShim/containerRegistryAuth.go:65]: Yup, Good catch. for some reason I removed that last night and don't recall why

File: cmd/k3d/create.go (hunk 1 above)
Comment (CristhianF7) [reply cmd/k3d/create.go:1206]: I'll remove the Stat since we have talked a lot about this 😄

**Review by patrickdappollonio — APPROVED** [review]

**Review by jokestax — APPROVED** [review]

**Pull request merged**